oshi-core/src/main/java/oshi/software/os/windows/WindowsInstalledApps.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2025 The OSHI Project Contributors
+ * SPDX-License-Identifier: MIT
+ */
+package oshi.software.os.windows;
+
+import oshi.driver.windows.registry.InstalledAppsData;
+import oshi.software.os.ApplicationInfo;
+import oshi.util.Constants;
+import oshi.util.ParseUtil;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class WindowsInstalledApps {
+
+    public static List<ApplicationInfo> queryInstalledApps() {
+        List<ApplicationInfo> appInfoList = new ArrayList<>();
+
+        List<Map<InstalledAppsData.AppInfo, String>> registryResults = InstalledAppsData.queryInstalledApps();
+
+        for (Map<InstalledAppsData.AppInfo, String> app : registryResults) {
+            String installDate = ParseUtil.getValueOrUnknown(app, InstalledAppsData.AppInfo.INSTALLDATE);
+            long installDateEpoch = 0;
+            if (!installDate.equals(Constants.UNKNOWN)) {
+                installDateEpoch = ParseUtil.parseDateToEpoch(installDate, "yyyyMMdd");
+            }
```

Comment:
This seems a strange pattern to create a new method to use UNKNOWN as a default and then check it, rather than just use `app.getOrDefault()` with an empty string (or even null) default that you can then check for with `Util.isBlank()` (or even `isBlankOrUnknown()`!).  Or just send it to the parse method and do some sanity checking of the format before parsing.  And offer a default from that method rather than defaulting to 0.